services/traction/api/main.py

```diff
@@ -79,6 +80,7 @@ async def on_tenant_startup():
     logger.warning(">>> Starting up app ...")
     subscribe_workflow_events()
     subscribe_protocol_listeners()
+    subscribe_job_list()
     subscribe_task_listeners()
     subscribe_webhook_events()
```

Comment:
i'd say that now we have the API "working" we can go over and address all of this. of course it can be (and should be) simpler. now is a great time to look at everything as a whole and optimize it. in my experience, refactoring to perfection is isolation is not a great idea. what is a nice little solution to a small problem doesn't mean it will work as nicely in the larger problem.  i don't think we really have any major problems to tackle except tightening up (a lot) the implementation. and now we can look at it as a complete package, with complete understanding of what exactly needs to happen where/when and by what actor and what implications we have to deal with that are out of our control (ie what does acapy do, what does the endorser do and how that affects our implementations). 

as far as string matching... 🤷 - I'm not sure there is another way when we are dealing with webhooks and events as the primary source of action? but let's put on the thinking hats and get to work.

Reply:
Why do we need Jobs? when we already have Tasks and Protocols? 

Just trying to understand the difference?

Reply:
there may not be a difference after refactoring... jobs were put in because we had no other way to loosely couple dependent tasks. and make issuer is dependent on public did which is dependent on endorser connection. tasks are one offs fired by traction - no real concern for state or transitions, protocols are listening and reacting to acapy webhooks, so follows state changes very closely, and has to have intricate knowledge of the payload to determine if they should be concerned about the message. protocol messages are very high level but are really specific to role and state. i don't know what the matrix is, but the 7 or 8 webhook message types represent > 50 events. the protocol classes are a way to key in on a specific event (ie issue credential/holder/credential_acked).

Reply:
Ah, so Jobs are that 'highest level' entity that can orchestra all other 'things' (acapy rfc's, tasks). That philosophically makes sense. Let me consider what that should be.

Reply:
jobs and "protocols" (as you said we should change that name if just remains a listener handler) are pretty close and could probably be handled by the same code. they do care about state transition and may need to do work as something goes through various states. tasks are simpler, but if jobs are coded a certain way, tasks could be absorbed in there too - just need basically the start function (the kick off work function) and that's about it. it's pretty reasonable to think that all 3 of these things can be addressed in one way.

Reply:
but i don't think that's the point of this PR to refactor protocol/task/job. it was to get that work out of v0 and explore the issue of async dependencies, how to effectively chain those async jobs